transform/all_collection/spark/transformations/commons.py

```diff
@@ -189,7 +216,43 @@ def map_publisher(df: DataFrame) -> DataFrame:
 
 def simplify_language(df: DataFrame) -> DataFrame:
     """Simplify language - get only label and convert structure to a string"""
-    return df.withColumn("language", col("language")["label"])
+    return df.withColumn(LANGUAGE, col(LANGUAGE)["label"])
+
+
+def map_language(df: DataFrame, harvested_properties: Dict) -> DataFrame:
+    """Harvest language and standardize its value"""
+    languages = df.select(LANGUAGE).collect()
+    language_column = []
+
+    for language in languages:
+        language = language[LANGUAGE]
+        if not language:
+            language_column.append(None)
+            continue
+
+        if isinstance(language, list):
+            language_row = []
+            for lang in language:
+                for desired_lan, old_lang in language_mapping.items():
+                    if lang in old_lang:
+                        language_row.append(desired_lan)
+                        break
+                else:
+                    language_row.append(lang)
+            language_column.append(language_row)
+
+        elif isinstance(language, str):
+            for desired_lan, old_lang in language_mapping.items():
+                if language.lower() in old_lang:
+                    language_column.append([desired_lan])
+                    break
+            else:
+                language_column.append([language])
+        else:
+            raise TypeError(f"{language} is not list or str")
```

Comment:
To deep nesting of `for` loops and `if`'s.

Reply:
You may rewrite it or leave it. :P

Reply:
I rewrote that